libcxx/include/__functional/move_only_function_impl.h

```diff
@@ -133,20 +117,20 @@ class _LIBCPP_MOVE_ONLY_FUNCTION_TRIVIAL_ABI move_only_function<_ReturnT(
   _LIBCPP_HIDE_FROM_ABI void __construct(_Args&&... __args) {
     static_assert(is_constructible_v<decay_t<_Func>, _Func>);
 
-    using _FuncWraps = __function_wrappers<_Func>;
-    using _UnRefFunc = remove_reference_t<_Func>;
+    using _FuncWraps  = __function_wrappers<_Func>;
+    using _StoredFunc = decay_t<_Func>;
 
     __call_ = &_FuncWraps::__call;
     if constexpr (!is_trivially_destructible_v<decay_t<_Func>>) {
       __destroy_ = &_FuncWraps::__destroy;
     }
 
     if constexpr (__fits_in_buffer<_Func>) {
-      std::construct_at(reinterpret_cast<_UnRefFunc*>(__buffer_.data()), std::forward<_Args>(__args)...);
+      std::construct_at(reinterpret_cast<_StoredFunc*>(__buffer_.data()), std::forward<_Args>(__args)...);
     } else {
       unique_ptr<std::byte[]> __ptr{
           static_cast<std::byte*>(::operator new(sizeof(_Func), std::align_val_t(alignof(_Func))))};
```

Comment:
This doesn't seem right; `unique_ptr<std::byte[]>` will call `delete [] p;`, but I don't _think_ that matches up with this arcane `::operator new` call.

I strongly recommend eliminating all dependencies on `std::byte`. (That'll also eliminate a header dependency, which is nice.) Then, at least in my version I just used `make_unique`, but I'm curious if you see a reason that wouldn't work correctly here. (And if so, please write a regression test that catches it!)

Reply:
I don't think using `make_unique` would be incorrect here, but the explicit alignment allows operator new a bit more freedom in what it returns. Why exactly are you recommending to remove dependencies on `std::byte`?

Reply:
> Why exactly are you recommending to remove dependencies on `std::byte`?

Coincidentally, Vittorio Romeo just posted something about `std::byte` the other day: https://vittorioromeo.info/index/blog/debug_performance_cpp.html ( https://godbolt.org/z/Gs4ehe8rK ). His beef is the _performance pessimization_ implied by `byte`'s extra layers of indirection. But my own beefs are different from Vittorio's, and twofold:
- the compile-time penalty for dragging in the definition of `std::byte` (including a new header is slow, unnecessary dependencies trigger Hyrum's Law, each new dependency theoretically makes it one straw more difficult to avoid circular dependencies)
- the philosophical idea that "core-language is better than library": `char` > `std::byte`, `struct` > `std::tuple`, `for` > `std::for_each`, `int[10]` > `std::array<int, 10>`, `alignof` > `std::alignment_of`, `alignas` > `std::aligned_storage`, `auto(x)` > `std::decay_t<X>(x)`, `::new` > `std::construct_at`, etc. Basically, every time the core language gives a way to do a task without library dependencies, one should use the core-language way _as a general rule_. Use library workarounds only when the core language hasn't (yet) provided a built-in solution. (This can be justified by appeal to runtime debug performance, or minimizing dependencies, or greater readability, or whatever; but at some point it just becomes a law in its own right. I rarely need to peek under the hood of this law, these days.)